elements/pfe-codeblock/src/pfe-codeblock.js

```diff
@@ -0,0 +1,269 @@
+      if (!this._codeblockContainer.textContent) {
+        this._codeblockRender.innerHTML = "";
+        return;
+      }
+
+      // TODO: when we stop supporting IE11, the need to disconnect and
+      // then reconnect will no longer be needed
+      if (window.ShadyCSS) {
+        observer.disconnect();
+      }
+
+      this.codeblock = this._codeblockContainer.textContent;
+
+      if (window.ShadyCSS) {
+        this._muationObserve();
+      }
+    });
+  }
+
+  //return class for line numbers
+  lineNumberCssClass() {
+    return this.codeLineNumbers ? " line-numbers" : "";
+  }
+
+  //return a valid prism.js language css class
+  codePrismLanguage() {
+    return "language-" + this.codeLanguage;
+  }
+
+  //return a prism.js language lib
+  codePrismLanguageLoad() {
+    return Prism.languages[this.codeLanguage];
+  }
+
+  //return applied classes for pre
+  appliedCssClasss() {
+    return this.codePrismLanguage() + this.lineNumberCssClass();
+  }
+
+  //apply classes to the markup for prism.js
+  setComponentClasses() {
+    this._codeblockRender.setAttribute("class", this.codePrismLanguage());
+    this._codeblockRenderOuterPreTag.setAttribute("class", this.appliedCssClasss());
+    if (this.codeLineNumberStart !== 1) {
+      this._codeblockRenderOuterPreTag.style.counterReset = "linenumber " + (this.codeLineNumberStart - 1);
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    //create dom elements and attach language styles
+    this._codeblockRenderOuterPreTag = document.createElement("pre");
+    this._codeblockRender = document.createElement("code");
+    this._codeblockRender.setAttribute("codeblock-render", "");
+    this.setComponentClasses();
+    this._codeblockRenderOuterPreTag.appendChild(this._codeblockRender);
+
+    //Add to shadow-root
+    //this.appendChild(this._codeblockRenderOuterPreTag);
+    this.shadowRoot.appendChild(this._codeblockRenderOuterPreTag);
+
+    this.shadowRoot.querySelector("slot").addEventListener("slotchange", () => {
+      if (!this._codeblockContainer) {
+        this._codeblockContainer = this.querySelector("[codeblock-container]");
+        this._codeblockContainer.style.display = "none";
+
+        this._init();
+      }
+    });
+
+    this.addEventListener(PfeCodeblock.events.change, this._changeHandler);
+  }
+
+  disconnectedCallback() {
+    this._observer.disconnect();
+    this.removeEventListener(PfeCodeblock.events.change, this._changeHandler);
+  }
+
+  // Process the attribute change
+  attributeChangedCallback(attr, oldValue, newValue) {
+    super.attributeChangedCallback(attr, oldValue, newValue);
+  }
+
+  _readyStateChangeHandler(event) {
+    if (event.target.readyState === "complete") {
+      document.removeEventListener("readystatechange", this._readyStateChangeHandler);
+      this._init();
+    }
+  }
+
+  _changeHandler(event) {
+    this.emitEvent(PfeCodeblock.events.change, {
+      detail: {}
+    });
+  }
+
```

Comment:
The _changeHandler above is emitting an event without any detail. Is this event necessary?